airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -163,6 +163,7 @@ class KubernetesPodOperator(BaseOperator):
         'config_file',
         'pod_template_file',
         'namespace',
+        'k8s_resources',
```

Comment:
@eladkal  I wasn't sure, so I went with the comment on the previous PR on this -- 

https://github.com/apache/airflow/pull/23531#issuecomment-1134181785

Reply:
I'm also perplexed as to why I'm getting failures on google auth credentials

Reply:
I don't think this was right?

What users are doing is:

```
compute_resources=k8s.V1ResourceRequirements(
    requests={
        'memory': '512Mi',
        'cpu': '500m'
    },
    limits={
        'memory': '1Gi',
        'cpu': 500m
    }
)

KubernetesPodOperator(..., container_resources=compute_resources)
```

But what they want to do is instead of specifying hard coded values like `512Mi` they want it to be templated so something like:

```
    requests={
        'memory': '{{ dag_run.conf["request_memory"] }} ',
        'cpu': '500m'
    }
```

note that the `k8s_resources` is not a field exposed to the users.
It is used mainly for backward compatibility where users used to pass resources in dict rather than `V1ResourceRequirements`:

https://github.com/apache/airflow/blob/53d68049d9bf4cec6b7d57545f15409dab0caed1/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py#L220-L228


I think we are also about to remove this support for `resources` in the next release of the provider so the `k8s_resources` will no longer be needed? cc @jedcunningham @dstandish

Reply:
@eladkal thanks for the detailed review.  I will get to this later today!

Reply:
Raised PR https://github.com/apache/airflow/pull/27197

Reply:
thanks @eladkal I will rebase and re-push this PR once that PR is merged.

Reply:
https://github.com/apache/airflow/pull/27197 is merged

Reply:
Huzzah!  @eladkal I'll rebase & re-push tomorrow.  thx :)